phpBB/includes/acp/acp_email.php

```diff
@@ -229,31 +217,50 @@ function main($id, $mode)
 				);
 				extract($phpbb_dispatcher->trigger_event('core.acp_email_send_before', compact($vars)));
 
+				$messenger = $phpbb_container->get('messenger.method_collection');
```

Comment:
I recommend adding a type hint for $messenger (\phpbb\di\service_collection)

Reply:
I'm not quite sure where to add a type hint here.

Reply:
Do you mean this?
```php
$messenger = (\phpbb\di\service_collection) $phpbb_container->get('messenger.method_collection');
```

Reply:
```php
/** @var service_collection */
$phpbb_container->get('messenger.method_collection');
```